### ISSUE TITLE
Cannot distinguish between EGIT_COMMIT and ${PN}_LIVE_COMMIT

### ISSUE THREAD

**Issue by doctaweeks** [mgorny/smart-live-rebuild#5]
It looks like the git-r3 eclass just internally sets EGIT_COMMIT when ${PN}_LIVE_COMMIT is used which means smart-live-rebuild can't distinguish between the two. In the latter case it may/should call out packages for updates as they really are live.

I suspect this is a limitation similar to issue #4.

**Comment by mgorny** [comment]
Actually, I think it's tad different. If you set `${PN}_LIVE_COMMIT` in make.conf, then obviously the package is bound to specific commit and there's no point in rebuilding it. If you change it or remove it, it's your responsibility to rebuild the package afterwards.

That said, that *_LIVE_COMMIT thing is backwards-compatibility stuff. Use EGIT_OVERRIDE_* variables instead. S-l-r isn't aware of them at all at the moment, so they will work the way you want them to.

**Comment by doctaweeks** [comment]
>  If you set ${PN}_LIVE_COMMIT in make.conf, then obviously the package is bound to specific commit and there's no point in rebuilding it.

Agreed. However, it could also be used on the command line during isolated testing and a user may still want it called out later when s-l-r is run.

> Use EGIT_OVERRIDE_* variables instead.

Thanks. I was not aware of these since they are not in the eclass reference or man pages. Reference for anyone else: https://archives.gentoo.org/gentoo-dev/message/947d684c2632a0208da9cf4704694f3d

**Comment by mgorny** [comment]
Yeah, I need to figure out how to document them properly.
